Files/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs

```diff
@@ -505,33 +507,53 @@ public async Task<ReturnResult> CopyItemAsync(IStorageItemWithPath source, strin
 
         public async Task<ReturnResult> CopyItemsFromClipboard(DataPackageView packageView, string destination, bool registerHistory)
         {
-            if (!packageView.Contains(StandardDataFormats.StorageItems))
+            if (packageView.Contains(StandardDataFormats.StorageItems))
             {
-                // Happens if you copy some text and then you Ctrl+V in Files
-                // Should this be done in ModernShellPage?
-                return ReturnResult.BadArgumentException;
-            }
+                IReadOnlyList<IStorageItem> source;
+                try
+                {
+                    source = await packageView.GetStorageItemsAsync();
+                }
+                catch (Exception ex) when ((uint)ex.HResult == 0x80040064 || (uint)ex.HResult == 0x8004006A)
+                {
+                    return ReturnResult.UnknownException;
+                }
+                ReturnResult returnStatus = ReturnResult.InProgress;
 
-            IReadOnlyList<IStorageItem> source;
-            try
-            {
-                source = await packageView.GetStorageItemsAsync();
-            }
-            catch (Exception ex) when ((uint)ex.HResult == 0x80040064 || (uint)ex.HResult == 0x8004006A)
-            {
-                return ReturnResult.UnknownException;
+                List<string> destinations = new List<string>();
+                foreach (IStorageItem item in source)
+                {
+                    destinations.Add(Path.Combine(destination, item.Name));
+                }
+
+                returnStatus = await CopyItemsAsync(source, destinations, registerHistory);
+
+                return returnStatus;
             }
-            ReturnResult returnStatus = ReturnResult.InProgress;
 
-            List<string> destinations = new List<string>();
-            foreach (IStorageItem item in source)
+            if (packageView.Contains(StandardDataFormats.Bitmap))
             {
-                destinations.Add(Path.Combine(destination, item.Name));
+                try
+                {
+                    var imgSource = await packageView.GetBitmapAsync();
+                    using var imageStream = await imgSource.OpenReadAsync();
+                    var folder = await StorageFolder.GetFolderFromPathAsync(destination);
+                    // Set the name of the file to be the current time and date
+                    var file = await folder.CreateFileAsync($"{DateTime.Now:mm-dd-yy-HHmmss}.png");
```

Comment:
If I were you I would be doing a filename check after generating the DateTime stamp to be use for the file name in event where the user may end up pressing ctrl + v in repetitive succession. If there was a case of a duplicated filename then it would be ideal to append a numeric index depending on the number of duplicates at the end of the filename, i.e. 01-23-2021 (1).

Reply:
Or just `var file = await folder.CreateFileAsync($"{DateTime.Now:mm-dd-yy-HHmmss}.png", CreationCollisionOption.GenerateUniqueName);`

Reply:
> Or just `var file = await folder.CreateFileAsync($"{DateTime.Now:mm-dd-yy-HHmmss}.png", CreationCollisionOption.GenerateUniqueName);`

Nice and simpler then 😁